acceptance/tests/pxp-module-puppet/run_puppet_agent_non_root.rb

```diff
@@ -0,0 +1,59 @@
+require 'pxp-agent/test_helper.rb'
+require 'yaml'
+
+test_name  'Run puppet agent as non-root' do
+
+  platform = agents.first.platform
+  skip_test "Test is not compatible with #{platform}" if platform =~ /windows/
+
+  step 'create non-root user on all nodes' do
+    on(agents, puppet("resource user foo ensure=present managehome=true"))
+
+    #TODO fix this only one agent ???
+    @user_home_dir = YAML.load(on(agents.first, puppet("resource user foo --to_yaml")).stdout)['user']['foo']['home']
+    @user_puppetlabs_dir = "#{@user_home_dir}/.puppetlabs"
+    @user_puppet_dir = "#{@user_puppetlabs_dir}/etc/puppet"
+    @user_pxp_dir = "#{@user_puppetlabs_dir}/etc/pxp-agent"
+    on(agents, "mkdir -p #{@user_puppet_dir}")
+    on(agents, "mkdir -p #{@user_pxp_dir}")
+  end
+
+  teardown do
+    on(agents, "kill -9 `pidof pxp-agent`", :accept_all_exit_codes => true)
```

Comment:
Will this work everywhere?

Reply:
At the least, pidof is not present on AIX.

Reply:
Can test against AIX 7.1 by:
- requesting an nspooler instance
- using it in `be rake ci:test:aio SHA=... TEST_TARGET=aix71-POWERa{hypervisor=none,hostname=<hostname>}`

Reply:
`get_process_pids` can probably be used as a replacement. It returns an array of matching PIDs.

Reply:
fixed

Reply:
Let me know when you've pushed the fix...

Reply:
pushed